admin/server/auth/handlers.go

```diff
@@ -53,6 +55,18 @@ func (a *Authenticator) authLogin(w http.ResponseWriter, r *http.Request) {
 	// Get auth cookie
 	sess, err := a.cookies.Get(r, cookieName)
 	if err != nil {
+		var cookieErr securecookie.Error
+		if errors.As(err, &cookieErr) && cookieErr.IsDecode() {
+			// clear cookie
+			sess.Options.MaxAge = -1
+			err = sess.Save(r, w)
+			if err != nil {
+				http.Error(w, fmt.Sprintf("failed to clear session: %s", err), http.StatusInternalServerError)
+				return
+			}
+			http.Redirect(w, r, r.URL.RequestURI(), http.StatusTemporaryRedirect)
+			return
+		}
```

Comment:
We have `a.cookies.Get` in several places where we need this check.

I think instead of clear+redirect, it should just set `sess` to an empty/fresh session if the cookie fails to parse. The cookie will usually be updated shortly after anyway. And maybe log the error?

Reply:
The usual [recommendation](https://github.com/gorilla/securecookie/blob/9865a2b6137913abf1982cc4b1e8acdc35b2f052/securecookie.go#L39) seems to be to reject the request completely. Hence I am clearing the cookie and redirecting to login page.
From what I understood this should only be happening in cases when user is trying to edit the cookies manually hence its okay to not log this error. The expired timestamp happened bcz of `max-age` configuration mismatch which is fixed now.
Since the recommendation is to reject the request, I am not taking any action anywhere and the client(UI in this case) will redirect the user to login page but we can redirect the user to login page in other cases as well.

Reply:
I agree, but currently it's not redirecting to the login page, it's redirecting to the same page (which I think makes sense). So in this case, what I'm talking about will be the exact same behavior, just without a redirect and extra return case.

Everywhere, we could just have `sess := a.getCookie(r, name)` (without the error case, since if there's an error, it will just log and return a clean cookie). Would be cleaner?

Reply:
Btw, this can also happen if we hard-reset session keys due to a compromise. Or if we reduce `max-age` in the future.